### PR TITLE
MC-12730: Added listing and get of brandings and knowledge bases

### PR DESCRIPTION
**Pull request by fyrnaga**
#### Changes made

- Added documentation for the branding list and get
- Added documentation for the knowledge base list and get

#### Related PRs
- []()

### CODE REVIEW

File: source/includes/administration/_branding.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          "content": "iVBOR...kJggg==",
+          "updatedAt": "2020-09-14T11:42:52.000Z"
+        }
+      ]
+    }
+  ]
+}
+```
+
+List the branding configured on the system.
```
Comment (dpmurray21) [inline source/includes/administration/_branding.md]: on the system -> for the organization ?

File: source/includes/administration/_branding.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Query Params | &nbsp;
+---- | -----------
+`organization_id`<br/>*UUID* | Return only branding for the provided organization id. If not provided, will default to the user organization.
```
Comment (dpmurray21) [inline source/includes/administration/_branding.md]: Return only the branding for the provided organization id. If not provided, will default to the user's organization.

File: source/includes/administration/_branding.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+Attributes | &nbsp;
+---------- | -----------
+`id`<br/>*UUID* | The id of the branding.
+`organization.id`<br/>*UUID* | The organization id that the branding is linked to.
+`defaultLanguage`<br/>*String* | The default language.
+`activeLanguages`<br/>*Array[String]* | List of the possible languages.
+`defaultTimezone`<br/>*String* | The default timezone.
+`applicationName`<br/>*String* | The application name displayed to the user.
+`factory`<br/>*boolean* | If the branding is the factory branding.
+`artifacts`<br/>*Array[Object]* | The list of artifacts link to the branding. This includes the custom css, colours css, logos, favicons.
```
Comment (dpmurray21) [inline source/includes/administration/_branding.md]: *linked to the branding*

File: source/includes/administration/_branding.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
```
Comment (dpmurray21) [inline source/includes/administration/_branding.md]: linked to the branding*

File: source/includes/administration/_knowledge_base.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    }
+  ]
+}
+```
+
+List the knowledge base configured on the system.
+
+Query Params | &nbsp;
+---- | -----------
+`organization_id`<br/>*UUID* | Return only knowledge base for the provided organization id. If not provided, will default to the user organization.
```
Comment (dpmurray21) [inline source/includes/administration/_knowledge_base.md]: Same as above

File: source/includes/administration/_knowledge_base.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+`categories.rank`<br/>*integer* | The category rank.
+`categories.icon`<br/>*string* | The category icon.
+`categories.createdAt`<br/>*string* | The creation date.
+`categories.updatedAt`<br/>*string* | The update date.
+`categories.featured`<br/>*boolean* | If the category needs to be displayed.
+`categories.translations`<br/>*Array[Object]* | The translation objects for the category.
+`categories.translations.id`<br/>*UUID* | The id of the translation.
+`categories.translations.language`<br/>*string* | The language of the translation.
+`categories.translations.text`<br/>*string* | The content of the translation.
+`categories.translations.type`<br/>*string* | The type of translation. Possible values are: title, body, description or url_slug.
```
Comment (dpmurray21) [inline source/includes/administration/_knowledge_base.md]: Category translations can only be of type "title"

Comment (dpmurray21) [reply source/includes/administration/_knowledge_base.md]: Unless this isn't for notification category?

File: source/includes/administration/_knowledge_base.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
```
Comment (dpmurray21) [inline source/includes/administration/_knowledge_base.md]: Same as above (Depending on what the categories object is)

File: source/index.html.md [hunk 8]
```diff
@@ -23,6 +23,8 @@ includes:
   - administration/identity_providers
   - administration/service_providers
   - administration/saml_settings
+  - administration/branding
+  - administration/knowledge_base
```
Comment (dpmurray21) [inline source/index.html.md:27]: Was I supposed to add this as well when adding the API docs for the notifications and categories?

File: source/includes/administration/_knowledge_base.md (hunk 6 above)
Comment (fyrnaga) [reply source/includes/administration/_knowledge_base.md]: it is not the same as the notification category, but put use the proper values

File: source/index.html.md (hunk 8 above)
Comment (fyrnaga) [reply source/index.html.md:27]: yep... otherwise it is not shown in the menu

File: source/includes/administration/_knowledge_base.md (hunk 7 above)
Comment (fyrnaga) [reply source/includes/administration/_knowledge_base.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_knowledge_base.md (hunk 6 above)
Comment (fyrnaga) [reply source/includes/administration/_knowledge_base.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_knowledge_base.md (hunk 5 above)
Comment (fyrnaga) [reply source/includes/administration/_knowledge_base.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_branding.md (hunk 4 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_branding.md (hunk 3 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_branding.md (hunk 2 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_branding.md (hunk 1 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: 2e431cc0d0e01031facb1b80afb966f6b042c0b7

File: source/includes/administration/_branding.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+`applicationName`<br/>*String* | The application name displayed to the user.
+`factory`<br/>*boolean* | If the branding is the factory branding.
+`artifacts`<br/>*Array[Object]* | The list of artifacts linked to the branding. This includes the custom css, colours css, logos, favicons.
+`artifacts.id`<br/>*UUID* | The artifact id.
+`artifacts.name`<br/>*string* | The artifact name.
+`artifacts.mimeType`<br/>*string* | The mime type of the artifact.
+`artifacts.type`<br/>*string* | The artifact type. The possibles values are: LOGO, LOGO_SQUARE, FAVICON, CUSTOM_CSS, COLORS.
+`artifacts.content`<br/>*string* | The artifact content.
+`artifacts.createdAt`<br/>*string* | The artifact creation date.
+`artifacts.updatedAt`<br/>*string* | The artifact update date.
```
Comment (HassanHaidar) [inline source/includes/administration/_branding.md]: Some string fields are String and others are string. Is there a difference?

File: source/includes/administration/_knowledge_base.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+`categories.articles.rank`<br/>*integer* | The article rank.
+`categories.articles.gitKey`<br/>*string* | The article git key.
+`categories.articles.createdAt`<br/>*string* | The creation date.
+`categories.articles.updatedAt`<br/>*string* | The update date.
+`categories.articles.translations`<br/>*Array[Object]* | The translation objects for the article.
+`categories.articles.translations.id`<br/>*UUID* | The id of the translation.
+`categories.articles.translations.language`<br/>*string* | The language of the translation.
+`categories.articles.translations.text`<br/>*string* | The content of the translation.
+`categories.articles.translations.type`<br/>*string* | The type of translation. Possible values are: title, body or url_slug.
+`categories.articles.translations.version`<br/>*integer* | The version of the translation.
```
Comment (HassanHaidar) [inline source/includes/administration/_knowledge_base.md:139]: Some strings are String and others are string

Comment (HassanHaidar) [reply source/includes/administration/_knowledge_base.md:139]: This applies to other files

File: source/includes/administration/_branding.md (hunk 9 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: i mistake on my part. I believe it should be all lower case.

**Review by HassanHaidar — COMMENTED** [review]
I left a comment about attribute fields. Other than that and @dpmurray21 's comments, it looks good!

File: source/index.html.md (hunk 8 above)
Comment (dpmurray21) [reply source/index.html.md:27]: Opened a PR for it, it will be merged shortly

**Review by dpmurray21 — APPROVED** [review]

File: source/includes/administration/_knowledge_base.md (hunk 10 above)
Comment (fyrnaga) [reply source/includes/administration/_knowledge_base.md:139]: c3c36bd6e3473d33872c5350f060c6bdb2938a06

File: source/includes/administration/_branding.md (hunk 9 above)
Comment (fyrnaga) [reply source/includes/administration/_branding.md]: c3c36bd6e3473d33872c5350f060c6bdb2938a06

**Review by HassanHaidar — APPROVED** [review]
Looks good!

**Pull request merged**